site/cgi-bin/pdf_check.cgi

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python3
+
+"""
+Given a filename, try to open that file, which should contain a JSON object
+containing a username and password, then test that against PAM printing out
+a JSON object that is authenticated and is true or false
+"""
+import cgi
+# If things are not working, then this should be enabled for better troubleshooting
+# import cgitb; cgitb.enable()
+import json
+import os
+import subprocess
+import shutil
+import stat
+
+# from grade_item.py
+def add_permissions(item,perms):
+    if os.getuid() == os.stat(item).st_uid:
+        os.chmod(item,os.stat(item).st_mode | perms)
+
+def add_permissions_recursive(top_dir,root_perms,dir_perms,file_perms):
+    for root, dirs, files in os.walk(top_dir):
+        add_permissions(root,root_perms)
+        for d in dirs:
+            add_permissions(os.path.join(root, d),dir_perms)
+        for f in files:
+            add_permissions(os.path.join(root, f),file_perms)
+
+print("Content-type: text/html")
+print()
+
+valid = True
+message = "Something went wrong."
+
+try:
+    arguments = cgi.FieldStorage()
+    num = int(os.path.basename(arguments['num'].value))
+    sem = os.path.basename(arguments['sem'].value)
+    course = os.path.basename(arguments['course'].value)
+    g_id = os.path.basename(arguments['g_id'].value)
+    ver = os.path.basename(arguments['ver'].value)
+
+    current_path = os.path.dirname(os.path.realpath(__file__))
+
+    uploads_path = os.path.join("/var/local/submitty/courses",sem,course,"uploads")
+    bulk_path = os.path.join("/var/local/submitty/courses",sem,course,"uploads/bulk_pdf",g_id,ver)
+    split_path = os.path.join("/var/local/submitty/courses",sem,course,"uploads/split_pdf",g_id,ver)
+
+    # copy folder
+    if not os.path.exists(split_path):
+        os.makedirs(split_path)
+
+    # adding permissions for the PHP
+    add_permissions_recursive(uploads_path, stat.S_IWGRP | stat.S_IXGRP, stat.S_IWGRP | stat.S_IXGRP, stat.S_IWGRP)
+
+    # copy over files to new directory
+    for filename in os.listdir(bulk_path):
+        shutil.copyfile(os.path.join(bulk_path, filename), os.path.join(split_path, filename))
+
+    # move to copy folder
+    os.chdir(split_path)
+
+    for filename in os.listdir(bulk_path):
+
+        total_pages = subprocess.check_output("pdftk " + filename + " dump_data | awk '/NumberOfPages/{print $2}'", shell=True)
+        total_pages = int(total_pages.decode('utf-8').rstrip())
+
+        #make sure # of pages is divisible
+        if (total_pages % num != 0):
+            valid = False
+            message = "Total # of pages: {t} is not divisible by the # of page(s) per exam: {n}".format(t=total_pages,n=num)
```

Comment:
Right now, the behavior is that all files uploaded will be processed, but as soon as one has an invalid # of pages the loop breaks and all split files under that timestamp upload are deleted.